projects/client-iframe/src/connector.ts

```diff
@@ -0,0 +1,71 @@
+import { ClientConnector, isHandshake, connectClient, applyApi, Client } from '@remixproject/plugin/connector'
+import { PluginClient } from '@remixproject/plugin/client'
+import { Message } from '../../utils/src/types/message'
+import { RemixApi } from '../../utils/src/api/remix-profile'
+import { Api, ApiMap } from '../../utils/src/types/api'
+import { checkOrigin } from './origin'
+import { listenOnThemeChanged } from './theme'
+
+
+export class IframeConnector implements ClientConnector {
+  source: Window
+  origin: string
+
+  constructor(private client: PluginClient) {}
+
+  /** Send a message to the engine */
+  send(message: Partial<Message>) {
+    if (this.source) {
+      this.source.postMessage(message, this.origin)
+    } else if (isHandshake(message)) {
+      window.parent.postMessage(message, '*')
+    }
+  }
+
+  /** Get messae from the engine */
+  on(cb: (message: Partial<Message>) => void) {
+    window.addEventListener('message', async (event: MessageEvent) => {
+      if (!event.source) throw new Error('No source')
+      // Check that the origin is the right one
+      const devMode = this.client.options.devMode
```

Comment:
If you need to pass `client` only for getting `devMode`, it would be better to either pass only `devMode` or `options`, so we really specify what's needed.
other way would be to create an env variable and use it directly from there.